packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -205,4 +204,14 @@ export class SelectedNetworkController extends BaseController<
 
     return networkProxy;
   }
+
+  setPerDomainNetwork(enabled: boolean) {
+    this.update((state) => {
+      state.perDomainNetwork = enabled;
+      return state;
+    });
+    Object.entries(this.state.domains).forEach(([domain, networkClientId]) => {
+      this.setNetworkClientIdForDomain(domain, networkClientId);
+    });
```

Comment:
Do we want to do this if `enabled` is false?

Reply:
ah yeah I missed that part! thank you! adding that now.